chore(kgo): add version support matrix (#7325)
Kong/docs.konghq.com: app/_src/gateway-operator/support.md

{:.note}
> {{site.kgo_product_name}} is a recently released product and does not currently provide an LTS version.

## Supported versions

{% assign latest_kgo_release = site.data.kong_latest_kgo.release %}
{% assign latest_kgo_eol = "Sept 2024" %}

<table style="display:table" width="100%">
<thead>
<tr>

>   <th>Kubernetes Version</th>

Reviewer: I agree that we should remove this column from the support page. Please make sure https://deploy-preview-7325--kongdocs.netlify.app/gateway-operator/latest/reference/version-compatibility/ is linked and very easy to find from this page

{:.note}
> {{site.kgo_product_name}} is a recently released product and does not currently provide an LTS version.

{% support_policy kong_versions_kgo%}

> *Table 1: Version Support for {{site.kgo_product_name}}*

{% include /md/support-policy.md %}

## Version compatibility with Kubernetes

> Version compatibility matrix with Kubernetes versions can be found in our [reference](/gateway-operator/{{ page.release }}/reference/version-compatibility/).

Reviewer 2: ```suggestion
You can see the version compatibility matrix with Kubernetes versions in the [compatibility reference](/gateway-operator/{{ page.release }}/reference/version-compatibility/).
```
